crypto/src/ed25519.rs

```diff
@@ -287,25 +338,15 @@ impl AggregateAuthenticator for Ed25519AggregateSignature {
         if pks.len() != messages.len() || messages.len() != sigs.len() {
             return Err(signature::Error::new());
         }
-        let mut inner_messages: Vec<&[u8]> = Vec::new();
-        for i in 0..messages.len() {
-            for _ in 0..pks[i].len() {
-                inner_messages.push(messages[i]);
+        let mut batch = batch::Verifier::new();
+
+        for i in 0..pks.len() {
+            for j in 0..pks[i].len() {
```

Comment:
rewrote this a bit to adapt to the library, since the length is the same, a double for loop should work? cc @punwai

Reply:
lgtm